src/main/java/org/jabref/gui/importer/GrobidPreferenceDialogHelper.java

```diff
@@ -7,29 +7,27 @@
 /**
  * Metadata extraction from PDFs and plaintext works very well using Grobid, but we do not want to enable it by default
  * due to data privacy concerns.
- * To make users aware of the feature, we ask each time before querying Grobid, giving the option to opt-out.
+ * To make users aware of the feature, we ask before querying Grobid, saving the users' preference for the future.
  */
-public class GrobidOptInDialogHelper {
+public class GrobidPreferenceDialogHelper {
 
     /**
-     * If Grobid is not enabled but the user has not explicitly opted-out of Grobid, we ask for permission to send data
-     * to Grobid using a dialog and giving an opt-out option.
+     * If the user has not explicitly opted-in/out of Grobid, we ask for permission to send data to Grobid by using
+     * a dialog. The users' preference is saved.
      *
      * @param dialogService the DialogService to use
      * @return if the user enabled Grobid, either in the past or after being asked by the dialog.
      */
     public static boolean showAndWaitIfUserIsUndecided(DialogService dialogService, GrobidPreferences preferences) {
-        if (preferences.isGrobidEnabled()) {
-            return true;
+        if (preferences.isGrobidPreference()) {
+            return preferences.isGrobidEnabled();
         }
-        if (preferences.isGrobidOptOut()) {
-            return false;
-        }
-        boolean grobidEnabled = dialogService.showConfirmationDialogWithOptOutAndWait(
+        boolean grobidEnabled = dialogService.showConfirmationDialogAndWait(
                 Localization.lang("Remote services"),
                 Localization.lang("Allow sending PDF files and raw citation strings to a JabRef online service (Grobid) to determine Metadata. This produces better results."),
-                Localization.lang("Do not ask again"),
-                optOut -> preferences.setGrobidOptOut(optOut));
+                Localization.lang("Yes"),
```

Comment:
Yes no only dialogs should be avoided, as the buttons are not self explanatory, better wording:

- Send to Grobid
- Do not send

Reply:
We have this requirement at https://devdocs.jabref.org/code-howtos/ui-recommendations.html#name-your-buttons-for-the-actions, but need more text for explanation. 

„yes“ is not an action.